src/pykeen/losses.py

```diff
@@ -282,13 +293,37 @@ def __init__(self, reduction: str = "mean"):
         super().__init__(reduction=reduction)
         self._reduction_method = _REDUCTION_METHODS[reduction]
 
+    # TODO: mypy does not seem to like this annotation
+    # @abc.abstractmethod
+    def forward(self, x: FloatTensor, target: FloatTensor, weight: FloatTensor | None = None) -> FloatTensor:  # noqa: DAR401
```

Comment:
I am not yet sure if this is a good abstraction: it seems to be mostly in line with the LCWA interpretation of score + label, and does not really fit well with e.g. margin-based ones.